### PR TITLE
Cypress visual testing integration

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

Added Cypress Visual Testing plugin. With this, we can write tests that can capture snapshots of the page visually while testing.

## Related Issue
Closes PWA-1661

### Verification Stakeholders
@dpatil-magento 

### Verification Steps
1. Checkout the branch locally.
2. Navigate to the integration tests package.
3. Run `yarn install`
3. Run `yarn test` to start the cypress UI.
4. Run the sample visual test (pageBuilder/homePage.spec.js). Other tests you might see are dummy ones, ignore them for now.

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: venia-integration-tests/src/tests/pageBuilder/homePage.spec.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/// <reference types="cypress" />
```
Comment (supernova-at) [inline venia-integration-tests/src/tests/pageBuilder/homePage.spec.js:1]: Is this needed?

File: venia-integration-tests/src/tests/pageBuilder/homePage.spec.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+context('Assert Venia Home Page pagebuilder content', () => {
+    describe('Assert Page Builder content', () => {
+        it('should render page builder content correctly', () => {
+            cy.visit('/');
+
+            cy.wait(3000);
```
Comment (supernova-at) [inline venia-integration-tests/src/tests/pageBuilder/homePage.spec.js:8]: I feel like this is too short. A no-cache, hard refresh page load on https://develop.pwa-venia.com/ took 6000ms for me.

Maybe 10000 here?

Comment (davemacaulay) [reply venia-integration-tests/src/tests/pageBuilder/homePage.spec.js:8]: https://docs.cypress.io/faq/questions/using-cypress-faq#How-can-I-make-Cypress-wait-until-something-is-visible-in-the-DOM

**Comment by supernova-at** [comment]
I got a couple errors while validating:

```
Error: The plugins file is missing or invalid.

Your `pluginsFile` is set to `path/to/pwa-studio/venia-integration-tests/src/plugins/index.js`, but either the file is missing, it contains a syntax error, or threw an error when required. The `pluginsFile` must be a `.js`, `.ts`, or `.coffee` file.

Or you might have renamed the extension of your `pluginsFile`. If that's the case, restart the test runner.

Please fix this, or set `pluginsFile` to `false` if a plugins file is not necessary for your project.
```

The file is definitely there and doesn't contain a syntax error. I'm not sure why / how it would throw an error when required.

<hr />

I also got an "update" message immediately from Cypress saying `7.1.0` is out (we're on `6.8.0`). Since we're just introducing Cypress I feel like we should be on the latest major version, at least. Do we have a reason we're not?

**Review by supernova-at — CHANGES_REQUESTED** [review]
Verification errors

**Comment by revanth0212** [comment]
> I also got an "update" message immediately from Cypress saying `7.1.0` is out (we're on `6.8.0`). Since we're just introducing Cypress I feel like we should be on the latest major version, at least. Do we have a reason we're not?

I have upgraded Cypress in this PR. Can you try `yarn install` in the folder instead of the outside which is `pwa-studio`? This package is not hooked up into yarn workspaces so doing a `yarn install` on the mono repo is not gonna install or update this package. 

This was done on purpose because this package is just like docs, has nothing to do with pwa-studio from the outside world's perspective.

File: venia-integration-tests/src/tests/pageBuilder/homePage.spec.js (hunk 2 above)
Comment (revanth0212) [reply venia-integration-tests/src/tests/pageBuilder/homePage.spec.js:8]: Yeah, you guys are right, but in the scope of this PR, I was working on introducing the plugin and integrating it not dint quite get into the test itself. In the following tickets, we can delegate this to the developer working on the ticket about what they think is the right choice of wait or use a helper as Dave suggested.

The test in this PR is a dummy one, just there to prove that this plugin works after the integration.

**Review by supernova-at — DISMISSED** [review]
Worked great once I installed everything properly 😅

**Comment by anthoula** [comment]
@revanth0212 Observed that the sample visual test (pageBuilder/homePage.spec.js) is not passing, and there is a failure at `dangerfile.js`
Screen Shot 2021-04-28 at 3 02 37 PM

**Comment by revanth0212** [comment]
I have made some changes to the comparison logic and hopefully, it will work across machines this time.
 
image

File: venia-integration-tests/src/tests/pageBuilder/homePage.spec.js (hunk 1 above)
Comment (revanth0212) [reply venia-integration-tests/src/tests/pageBuilder/homePage.spec.js:1]: Yeah, it is needed for IntelliSense to suggest cypress auto completions.

**Comment by dpatil-magento** [comment]
@revanth0212 It still fails when screen shot captured window size differs current running window size.

File: venia-integration-tests/src/plugins/resizeSnapshotPlugin.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+const jimp = require('jimp');
+const fs = require('fs');
+const { matchImageSnapshotPlugin } = require('cypress-image-snapshot/plugin');
+
+const resizeSnapshot = imageConfig => {
```
Comment (jimbo) [inline venia-integration-tests/src/plugins/resizeSnapshotPlugin.js:5]: Is this the function that resolves differences in terms of resolution and pixel density?

File: venia-integration-tests/src/support/__tests__/loadFullPageCommand.spec.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+            add: mockAdd
+        }
+    };
+});
+
+afterEach(() => {
+    delete global.Cypress;
+});
+
+test('sup', () => {
```
Comment (jimbo) [inline venia-integration-tests/src/support/__tests__/loadFullPageCommand.spec.js:15]: Can this test be skipped?

File: venia-integration-tests/src/plugins/resizeSnapshotPlugin.js (hunk 3 above)
Comment (revanth0212) [reply venia-integration-tests/src/plugins/resizeSnapshotPlugin.js:5]: Yes, that's the one.

**Comment by anthoula** [comment]
Tests pass. QA approved. Back @revanth0212 to confirm if there will be additional commits

**Review by dpatil-magento — APPROVED** [review]

**Pull request merged**